pkg/notify/notifier/pushover/pushover.go

```diff
@@ -0,0 +1,254 @@
+package pushover
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kubesphere/notification-manager/pkg/apis/v2beta2"
+	"github.com/kubesphere/notification-manager/pkg/async"
+	"github.com/kubesphere/notification-manager/pkg/notify/config"
+	"github.com/kubesphere/notification-manager/pkg/notify/notifier"
+	"github.com/kubesphere/notification-manager/pkg/utils"
+	"github.com/prometheus/alertmanager/template"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	DefaultSendTimeout = time.Second * 3
+	URL                = "https://api.pushover.net/1/messages.json"
+	DefaultTemplate    = `{{ template "nm.default.text" . }}`
+	MessageMaxLength   = 1024
+)
+
+type Notifier struct {
+	notifierCfg  *config.Config
+	pushover     []*config.Pushover
+	timeout      time.Duration
+	logger       log.Logger
+	template     *notifier.Template
+	templateName string
+	client       http.Client
```

Comment:
It needs a global HTTP client for all `pushover Notifier`, not for one.

Reply:
Like this? https://github.com/txfs19260817/notification-manager/blob/cb474dfd005a16ac5f25748f95a0acca1d320f52/pkg/notify/notifier/pushover/pushover.go#L31